src/NexusMods.Paths/AbsolutePath.cs

```diff
@@ -84,19 +85,34 @@ internal AbsolutePath(string? directory, string fileName, IFileSystem fileSystem
     /// </remarks>
     public static AbsolutePath FromFullPath(string fullPath, IFileSystem? fileSystem = null)
     {
-        var index = fullPath.LastIndexOf(Path.DirectorySeparatorChar);
-        if (index < 0)
-        {
-            // No directory
+        var span = fullPath.AsSpan();
+
+        // path is not rooted
+        var rootLength = GetRootLength(span);
+        if (rootLength == 0)
             return new AbsolutePath(null, fullPath, fileSystem ?? Paths.FileSystem.Shared);
+
+        // path is only the root directory
+        if (span.Length == rootLength)
+            return new AbsolutePath(fullPath, "", fileSystem ?? Paths.FileSystem.Shared);
+
+        var slice = span[rootLength..];
+        if (slice[^1] == PathSeparatorForInternalOperations)
+            slice = slice[..^1];
+
+        var separatorIndex = slice.LastIndexOf(PathSeparatorForInternalOperations);
+        if (separatorIndex == -1)
+        {
+            // root directory (eg: "/" or "C:\\") is the directory
+            return new AbsolutePath(span[..rootLength].ToString(), slice.ToString(), fileSystem ?? Paths.FileSystem.Shared);
         }
 
-        // Windows: "C:\foo", directory should be "C:"
-        // Linux: "/foo", directory should be "/"
-        var directory = index == 0 ? DirectorySeparatorCharStr : fullPath[..index];
+        // everything before the separator
+        var directorySpan = span[..(rootLength + separatorIndex)];
```

Comment:
Make sure to `SliceFast` here too. It should be safe as we handle the case of trailing slash above; but the JIT isn't smart enough to know that and inserts unnecessary bounds check.